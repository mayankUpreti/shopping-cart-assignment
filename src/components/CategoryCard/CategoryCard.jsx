import React from 'react'
import { useHistory } from 'react-router'

const CategoryCard=({data,order})=>{
const history=useHistory()
    return(
      <div className={` d-flex ${order%2==0?'flex-row-reverse':'flex-row'} justify-content-around border-bottom my-2 py-3`} >
      <div className=' align-center w-50'>
      <img className='category-image' src={data.imageUrl} alt=''/>
      </div>
        <div className=' align-center flex-column '>
            <h3 className='text-center font-weight-bold'>{data.name}</h3>
            <h5 className='text-center'>{data.description}</h5>
            <button id='product-category' role='button' className='btn btn-primary ' onClick={()=>history.push(`/products`)} >Explore {data.key}</button>
        </div> 
      </div>
    )

}


export default CategoryCard