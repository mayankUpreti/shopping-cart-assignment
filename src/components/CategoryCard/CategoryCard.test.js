import React  from 'react'
import CategoryCard from './CategoryCard'
import  {shallow} from 'enzyme';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { createMemoryHistory } from 'history'
import { Router } from 'react-router-dom';
import '@testing-library/jest-dom'
it('testing a custom carousel ',()=>{
   const mockDb=[{
       id:1212,
       key:'ssdsd',
       name:'Tide',
       description:'a surf is a exccel',
       bannerImageUrl:'',
       bannerImageAlt:'yoyo'
   }]
    expect(shallow(<CategoryCard data={mockDb} order={1}/>)).toMatchSnapshot()

})


test('triggers path change', () => {
    const history = createMemoryHistory();
    const mockDb=[{
        id:1212,
        key:'ssdsd',
        name:'Tide',
        description:'a surf is a exccel',
        bannerImageUrl:'',
        bannerImageAlt:'yoyo'
    }] 
    render(
      <Router history={history}>
        <CategoryCard data={mockDb} order={1}/>
      </Router>
    );
  
    const productcategory = screen.getByRole('button');
    expect(productcategory).toBeInTheDocument();
  
    userEvent.click(productcategory);
    expect(history.location.pathname).toBe('/products');
  });