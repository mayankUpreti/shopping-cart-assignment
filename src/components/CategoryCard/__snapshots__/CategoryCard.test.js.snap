// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing a custom carousel  1`] = `
<div
  className=" d-flex flex-row justify-content-around border-bottom my-2 py-3"
>
  <div
    className=" align-center w-50"
  >
    <img
      alt=""
      className="category-image"
    />
  </div>
  <div
    className=" align-center flex-column "
  >
    <h3
      className="text-center font-weight-bold"
    />
    <h5
      className="text-center"
    />
    <button
      className="btn btn-primary "
      id="product-category"
      onClick={[Function]}
      role="button"
    >
      Explore 
    </button>
  </div>
</div>
`;
